rfcs/text/0000-conceptual-data-model.md

```diff
@@ -0,0 +1,111 @@
+
+![](../images/0000-new-analysis-graph.png)
+
+Figure 3: Graph with a new analysis:  In this case, a second, independent analysis is run.  This might be the case when a new different analysis algorithm is being used.  The head graph includes both analysis results.   The provenance graph is obtained by reverse traversing the edges shown in red.
+
+
+![](../images/0000-reanalysis-graph.png)
+
+Figure 4: Graph with an upstream consequential metadata change that triggers a reanalysis.  This is the case where there is an error in a metadata field that is used to parameterize the analysis. Here the sequencing didn't change, however, the downstream analysis was triggered by a change in the biomaterial metadata (B1).  The provenance graph reverse traverses the edges shown in red, going to B1.2 even though the process metadata P1.1 has not changed.
+
+
+![](../images/0000-amended-graph.png)
+
+Figure 5: Graph with an upstream inconsequential metadata change that does not trigger a reanalysis.  Correcting natural-language descriptions or migrating the metadata schema are examples of this type of update.  Reverse-traversing the read edges gives the amended provenance graph.
+
+The *Project* metadata entity is as a container for all metadata and data that is part of the project.  All data in the DCP belongs to at least one project.  A given metadata instance may be associated with more than one project, such as a sample sequenced by multiple projects. Derived projects can be created for an analysis that used data from multiple projects.
+
+### Example use cases
+These sections collect various use cases to illustrate how they are represented or are problematic for this model.
+
+- Contributed discovers the wrong FASTQ was uploaded or a FASTQ is missing.  An update to the primary bundle is submitted with a replacement or new FASTQ.  This is a consequential change to analysis and will result in a new provenance graph.
+
+- Change in B1 in a metadata field that is used to parameterize the analysis so analysis is triggered but the outputs (F3) from analysis & re-analysis checksums are identical.  It would be desirable for the user to know the results are not changed with the new parameters.  This is not currently representable.  The users would have to detect themselves.
+
+- The similar case of updated software or parameters (P2) causes re-analysis but the outputs have identical checksums.
+
+
+### Unresolved Questions
+
+- Does this present a clear and rigorous model for interpreting multiple version of data for developers and advanced users? 
+
+- How should the re-analysis not change results be modeled?
+
+- Is the graph external to the metadata entities, allowing them to be shared by multiple nodes in a global graph, or are the entity instances the nodes?  That is, are the entity instances attributes of the nodes or the nodes themselves?  The current implementation with links.json supports both models, however, this needs to be explicitly defined.
```

Comment:
I favour an external graph. Otherwise we get into a real tangle of constantly updating entities themselves to add and remove these links.

Reply:
@justincc this is just conceptual.  A model where entities are nodes can still store the links physically separate from the entity structure.  The issue here is if entities can be shared by multiple graphs.